lib/plugin/instance.rb

```diff
@@ -137,11 +136,15 @@ def full_admin_route
   end
 
   def any_settings?
-    return false if !configurable?
+    configurable? && plugin_settings.values.length.positive?
+  end
+
+  def has_only_enabled_setting?
+    plugin_settings.values.length == 1
```

Comment:
Could be just:

```suggestion
    plugin_settings.one?
```

Reply:
I was wondering if there would be some rails mischief for this :P